File: frontend/src/types/content-template.ts
export interface ContentTemplateSection {
  title: string;
  description: string;
  prompt_hint: string;
}

export interface ContentTemplate {
  id: string;
  content_type: string;
  name: string;
  description: string;
  sections: ContentTemplateSection[];
  doc_name_pattern: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

File: backend/models/content_template.py
from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import String, Text, Boolean, func
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import Mapped, mapped_column

from backend.database import Base


class ContentTemplate(Base):
    __tablename__ = "content_templates"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    content_type: Mapped[str] = mapped_column(String, unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    sections: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    doc_name_pattern: Mapped[str | None] = mapped_column(String, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    created_at: Mapped[datetime] = mapped_column(default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(default=func.now(), onupdate=func.now(), nullable=False)

